go/types/list.go

```diff
@@ -185,18 +219,22 @@ func (l List) IterAll(f listIterAllFunc) {
 	})
 }
 
+// Iterator returns a ListIterator which can be used to iterate efficiently over a list.
 func (l List) Iterator() ListIterator {
 	return l.IteratorAt(0)
 }
 
+// IteratorAt returns a ListIterator starting at index.
```

Comment:
You might want to say, "if index is out of bounds, returns an iterator which returns nil as its next value"